Lib/unittest/mock.py

```diff
@@ -888,6 +902,12 @@ def _get_child_mock(self, **kw):
                 klass = Mock
         else:
             klass = _type.__mro__[1]
+
+        if self._mock_sealed:
+            attribute = "." + kw["name"] if "name" in kw else "()"
+            mock_name = self._extract_mock_name() + attribute
+            raise AttributeError(mock_name)
```

Comment:
I thought the attribute error was meaningful enough here.

It will look like:

Setting the attribute:
`AttributeError('Cannot set mock.test1.test3.test4')`

Getting the attribute:
``AttributeError('mock.test1.test3.test4')`

Happy to change the get if you propose a new format